### PR TITLE
Add support for HEIC/HEIF/TIFF

### PR DESCRIPTION
**Pull request by pulsejet**
Convert all images to JPEG and downscale before passing to node

#233 #111

### CODE REVIEW

**Comment by marcelklehr** [comment]
@pulsejet Thanks a lot! Would you allow me to push some commits?

**Comment by pulsejet** [comment]
@marcelklehr of course! I think you already should be able to.

File: lib/Classifiers/Classifier.php [hunk 1]
```diff
@@ -173,9 +173,13 @@ private function getConvertedFilePath(Node $file): string {
 		$tmpname = $this->tempManager->getTemporaryFile('.jpg');
 
 		try {
-			// Downscale into a temporary JPEG file
+			// Convert to a temporary JPEG file optionally downscaling
 			$imagick = new \Imagick($path);
-			$imagick->scaleImage(4096, 4096, true);
+			$dimensions = $imagick->getImageGeometry();
+			if ($dimensions['width'] > 4096 || $dimensions['height'] > 4096) {
```
Comment (pulsejet) [inline lib/Classifiers/Classifier.php:179]: Good catch. Imagick before 3.0.0 used to leave the image untouched if it was smaller than the specified scaling; I didn't know the behavior had changed.

**Review by marcelklehr — APPROVED** [review]

**Comment by pulsejet** [comment]
Thank you @marcelklehr! 🎉

**Comment by opa-rudi** [comment]
Any news about this topic? I have a lot of HEIC images from an iPhone that are not passed to the AI.

**Comment by marcelklehr** [comment]
Hi @opa-rudi make sure your nextcloud image preview mechanism supports HEIC

**Pull request merged**